src/eval/operation.rs

```diff
@@ -615,6 +615,19 @@ impl<R: ImportResolver> VirtualMachine<R> {
 
                 match_sharedterm! {t, with {
                         Term::Record(record) => {
+                            // While it's certainly possible to allow mapping over
+                            // a record with a sealed tail, it's not entirely obvious
+                            // how that should behave. It's also not clear that this
+                            // is something users will actually need to do, so we've
+                            // decided to prevent this until we have a clearer idea
+                            // of potential use-cases.
+                            if let Some(record::SealedTail { label, .. }) = record.sealed_tail {
+                                return Err(EvalError::BlameError(
+                                    label,
```

Comment:
Currently the error for this is pretty unclear, as the label doesn't have a tag. I considered doing something like this:

```rust
if let Some(record::SealedTail { mut label, .. }) = record.sealed_tail {
    label.tag = String::from("mapping over a polymorphically sealed record is forbidden");
    // ...
```

but I couldn't find anywhere else where we explicitly set the `tag` of a label without using the `%tag%` primitive. Is there a reason for that, or would setting the `tag` here be acceptable?

Reply:
I think one reason is that it would erase potentially user-defined tags, at least in the general case (in that, a user could set an error message, and then use `contract.apply` on a polymorphic record contract).

Note that we have had only one kind of blame error until now, but there is no reason for that (well, beside avoiding code duplication, but we could pull out a blame error message out of the `BlameError` match case). If you need a specific variant because the error reporting is different, it makes sense that PolymorhicTailIllegalAccess or whatever shorter and better name you find has its own variant.

Reply:
In general the interface or `Lbl` for users is pretty poor: we could reflect better our error reporting infra and give rather access `notes` to allow multiple lines of information without erasing the previous tag. But I digress

Reply:
Okay, thanks for clarifying! I actually started hacking in a new variant while I was spiking but then did indeed get put off by the duplication. I'll take another look now that everything else seems to be working correctly at least. :grin: